### PR TITLE
Allow using credentials for codeservers in transcripts

### PR DESCRIPTION
**Pull request by ChrisPenner**
Previously I had disabled this, just out of concern that there's potential for abuse if someone were to send a transcript and convince someone to run it, but I think there's also the argument that users like using transcripts to interact with UCM, and it's reasonable for them to want to interact with codeservers as part of that too.

The more pressing reason for the change is that this allows us to use transcripts for testing our sync algorithms.

This PR:

* Allows transcripts to access user credentials for interacting with codeservers
* Allow overriding credentials with a specific access token using `UNISON_SHARE_ACCESS_TOKEN`; When provided this token will be used for all codeserver requests. this was much easier to wire up for automating transcript running in Enlil.
  * Most users will just use their standard credentials store, but since the local-server credentials might get cycled at any time it's nice to be able to programmatically specify a specific token.

### CODE REVIEW

File: lib/unison-util-relation/unison-util-relation.cabal [hunk 1]
```diff
@@ -20,8 +20,6 @@ library
       Unison.Util.Relation
       Unison.Util.Relation3
       Unison.Util.Relation4
-  other-modules:
```
Comment (ChrisPenner) [inline lib/unison-util-relation/unison-util-relation.cabal]: No idea why this disappeared 🤔

File: unison-cli/src/Unison/Auth/HTTPClient.hs [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-import Unison.Auth.Tokens (TokenProvider, newTokenProvider)
+import Unison.Auth.Tokens (TokenProvider)
 import Unison.Codebase.Editor.UCMVersion (UCMVersion)
 import Unison.Prelude
 import Unison.Share.Types (codeserverIdFromURI)
 import qualified Unison.Util.HTTP as HTTP
 
 -- | Newtype to delineate HTTP Managers with access-token logic.
-newtype AuthorizedHttpClient = AuthorizedHttpClient HTTP.Manager
+newtype AuthenticatedHttpClient = AuthenticatedHttpClient HTTP.Manager
```
Comment (ChrisPenner) [inline unison-cli/src/Unison/Auth/HTTPClient.hs:14]: Sorry for the little bit of PR noise, but I realize I used the wrong "Auth*" name here 😶

File: unison-cli/src/Unison/Codebase/TranscriptParser.hs [hunk 3]
```diff
@@ -71,6 +75,13 @@ import Prelude hiding (readFile, writeFile)
 terminalWidth :: Pretty.Width
 terminalWidth = 65
 
+-- | If provided, this access token will be used on all
+-- requests which use the Authenticated HTTP Client; i.e. all codeserver interactions.
+--
+-- It's useful in scripted contexts or when running transcripts against a codeserver.
+accessTokenEnvVarKey :: String
+accessTokenEnvVarKey = "UNISON_SHARE_ACCESS_TOKEN"
```
Comment (aryairani) [inline unison-cli/src/Unison/Codebase/TranscriptParser.hs:83]: Could you add a note to the PR description about when/how one might use this

**Review by aryairani — APPROVED** [review]
LGTM

**Comment by aryairani** [comment]
@ChrisPenner How can I properly create or get and supply the `UNISON_SHARE_ACCESS_TOKEN`? Let's add this sort of thing to `development.markdown`

**Pull request merged**